src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -7,42 +7,30 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
-import java.math.BigDecimal;
-import java.util.ArrayList;
-import java.util.Currency;
-import java.util.List;
+
 
 public class ProductListPageServlet extends HttpServlet {
     
-    private ProductDao product = new ArrayListProductDao();
+    public static ProductDao product;
+
+    public ProductListPageServlet() {
+        this.product = new ArrayListProductDao();
+    }
     
     
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-//        List<Product> list = new ArrayList();
-//        list.add(product.getProduct(5L));
-          //product.delete(12L);
-        request.setAttribute("products",product.findProducts() );
+        synchronized(this){
```

Comment:
This is a bit tricky.

Please check spec: https://download.oracle.com/otn-pub/jcp/servlet-3.0-fr-eval-oth-JSpec/servlet-3_0-final-spec.pdf?AuthParam=1572334946_7f4786ec37dbfbce193ba15d6cc4e3cd

And look through the docs:
https://books.google.by/books?id=Wf5ihBC7gucC&pg=PA185&lpg=PA185&dq=synchronize+doget+dopost&source=bl&ots=hVdR-3iUll&sig=ACfU3U0n0yOw3jKh8C2ZK5ml_n_WjYor3Q&hl=en&sa=X&ved=2ahUKEwiph6GY_MDlAhUDwsQBHULGCHwQ6AEwDnoECAkQAQ#v=onepage&q&f=false

https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/http/HttpServlet.html

https://books.google.by/books?id=YC0geTPQt24C&pg=PA266&lpg=PA266&dq=synchronize+doget+dopost&source=bl&ots=wcQeQF4Hjb&sig=ACfU3U2j2SZSGf1lkrsq_ory2Ke_lxMj8g&hl=en&sa=X&ved=2ahUKEwiph6GY_MDlAhUDwsQBHULGCHwQ6AEwEXoECAcQAQ#v=onepage&q&f=false